packages/api/src/utils/serdes.ts

```diff
@@ -66,3 +66,25 @@ export function fromU64StrOpt(u64Str: U64Str | undefined): U64 | undefined {
 export function toU64StrOpt(u64: U64 | undefined): U64Str | undefined {
   return u64 !== undefined ? toU64Str(u64) : undefined;
 }
+
+const GRAFFITI_HEX_LENGTH = 66;
+
+export function toGraffitiHex(utf8: string): string {
+  const hex = toHexString(new TextEncoder().encode(utf8));
+
+  if (hex.length > GRAFFITI_HEX_LENGTH) {
+    // remove characters from the end if hex string is too long
+    return hex.slice(0, GRAFFITI_HEX_LENGTH);
+  }
+
+  if (hex.length < GRAFFITI_HEX_LENGTH) {
+    // right-pad with zeros if hex string is too short
+    return hex.padEnd(GRAFFITI_HEX_LENGTH, "0");
+  }
+
+  return hex;
+}
+
+export function fromGraffitiHex(hex: string): string {
+  return new TextDecoder("utf8").decode(fromHexString(hex));
```

Comment:
is there an edgecase where a graffiti is passed in that throws here?

The case I'm thinking of is a graffiti with 31 alphanumeric characters, then an emoji. toGraffitiHex then chops off part of the emoji resulting in invalid utf8?

Might be safer to wrap this in a try/catch.

Reply:
> The case I'm thinking of is a graffiti with 31 alphanumeric characters, then an emoji. toGraffitiHex then chops off part of the emoji resulting in invalid utf8?

I have tested that scenario with `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa🐼` which results in hex `0x61616161616161616161616161616161616161616161616161616161616161f0` which is decoded to `aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa�`, so looks like it can't display the last character of the UFT-8 string but it does not throw an error. 

`fromGraffitiHex` will mostly throw an error if [`fromHexString`](https://github.com/ChainSafe/ssz/blob/b0e99993b86c669b0e9a95882c1e7c85fdd97075/packages/ssz/src/util/byteArray.ts#L17) fails


> Might be safer to wrap this in a try/catch.

Before the API input validation was basically non existent, you could provide any kind of graffiti and the beacon node would somehow make it work. Now we are definitely stricter, an invalid hex value for example would throw an error. I guess what we do not want is that proposing a block fails due to an invalid graffiti.

Should be add a try-catch which would return an empty string as fallback value for the graffiti? And maybe add a warning log message?

Reply:
Yeah, I think its preferable to allow a malformed graffiti for now. try/catch + warning sounds sufficient.

Seems that fromHexString is the only throwable thing. TextDecoder seems to default to a "non-fatal" mode: https://developer.mozilla.org/en-US/docs/Web/API/TextDecoder/TextDecoder

Reply:
Added try-catch during decoding, if an invalid hex string is provided we will now just use an empty string (`""`) as graffiti.

Regarding the warning, it is not possible to access the logger here and I do not want to add a `console.log`. Maybe it might anyway better if we do some basic validation of the graffiti on startup and show the warning there if the graffiti is malformed. If another client is interacting with the BN API we can not do much, it is up to that client to warn the user about the malformed graffiti, I think it should anyway not be the concern of the BN.

Reply:
Regarding the default return value, changed it now to just return the provided value if decoding fails. This should not cause any issue downstream as it is how the BN currently handles the graffiti (without any modification).